application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -177,9 +182,23 @@ private MessageCreateAction createModMessage(String reportReason,
             .setDescription(reportReason)
             .setColor(AMBIENT_COLOR)
             .build();
-        return modMailAuditLog.sendMessageEmbeds(reportedMessageEmbed, reportReasonEmbed)
-            .addActionRow(DiscordClientAction.Channels.GUILD_CHANNEL_MESSAGE.asLinkButton(
-                    "Go to Message", guild.getId(), reportedMessage.channelID, reportedMessage.id));
+
+        MessageCreateAction message =
+                modMailAuditLog.sendMessageEmbeds(reportedMessageEmbed, reportReasonEmbed)
+                    .addActionRow(DiscordClientAction.Channels.GUILD_CHANNEL_MESSAGE.asLinkButton(
+                            "Go to Message", guild.getId(), reportedMessage.channelID,
+                            reportedMessage.id));
+
+        Optional<Role> moderatorRole = guild.getRoles()
+            .stream()
+            .map(Role::getName)
+            .filter(configModGroupPattern)
+            .findFirst()
+            .map(name -> guild.getRolesByName(name, false).get(0));
```

Comment:
Is getting all roles and then getting the correct role really necessary? Why dont you just remove the mapping and filter with `role.getName()`?

Reply:
good point.
```java
.stream()
.filter(role -> configModGroupPattern.test(role.getName()))
.findFirst()
```

Reply:
Because configModGroupPattern could match multiple roles in the future (in which case we will not get a single role anymore but a list of them and mention them all)

Reply:
But then my idea still works and the findFirst is causing the issue

Reply:
Yeah I see it, will change.

Reply:
Great thank you